### DISCUSSION TITLE
Document how you are using axe

### DISCUSSION THREAD

**Discussion by mgifford** [cds-snc/scan-websites#258] [Q&A]
How is what you are building compare with:

- https://github.com/alphagov/accessibility-monitoring
- https://github.com/GovTechSG/purple-hats or https://github.com/CivicActions/purple-hats
- https://github.com/MSU-NatSci/DomainAccessibilityAudit
- https://github.com/benbalter/Site-Inspector

Does it crawl the site or just review the home page?

**Comment by mohdnr** [comment]
Hi @mgifford sorry for the delay in getting back to you! We're currently using [scrapy](https://github.com/scrapy/scrapy) to crawl each page and then we're running axe-core against all the results and then pulling them into our reporting dashboard. Example below. 

The main different between this product and the ones listed above is that we're also performing security scans against websites to help identify vulnerabilities as new features are developed.

An architecture diagram is on the list of things we need to do!

![image](https://user-images.githubusercontent.com/85885638/145899260-9047c970-2a68-45c2-a6a3-6a8a82b9526b.png)

**Reply by mgifford** [reply]
I hadn't actually seen the shift from issue to discussion before. Neat.

Thought you'd also like this addition, which is also supporting PDF scans:
https://github.com/accessibility-luxembourg

Also worth checking out http://verapdf.org/

When building these out, Axe is definitely the default. I would recommend thinking about how the architecture would support swapping out Alfa or IBM's Equal Access engine too.  Different approaches & rules are interesting. Even just Google Lighthouse (which uses axe) might provide insights that would affect CX.

**Reply by mohdnr** [reply]
Thanks for the links, those look interesting. Our current architecture has all the scans triggering by step functions or sns topics with everything running in a docker container. This will make it easy for us to introduce new tools or to swap things out since there's no hard dependency.

We'll most definitely take those into consideration as we continue to build this tool.

**Reply by mgifford** [reply]
I've done much of the work if setting up this scanner within a Docker container https://github.com/CivicActions/purple-hats but I'm not quite there yet. You can only run https://github.com/MSU-NatSci/DomainAccessibilityAudit within Docker (but it's running a much older version of axe).

**Comment by maxneuvians** [comment]
The other thing worth noting is that we can scan modern JS web apps, so anything that uses react or angular is also in scope.
